java/vector/src/main/java/org/apache/arrow/vector/ipc/message/ArrowFooter.java

```diff
@@ -96,17 +126,40 @@ public Schema getSchema() {
     return recordBatches;
   }
 
+  public Map<String, String> getMetaData() {
+    return metaData;
+  }
+
   @Override
   public int writeTo(FlatBufferBuilder builder) {
     int schemaIndex = schema.getSchema(builder);
     Footer.startDictionariesVector(builder, dictionaries.size());
     int dicsOffset = writeAllStructsToVector(builder, dictionaries);
     Footer.startRecordBatchesVector(builder, recordBatches.size());
     int rbsOffset = writeAllStructsToVector(builder, recordBatches);
+
+    int metaDataOffset = 0;
+    if (metaData != null) {
+      int[] metadataOffsets = new int[metaData.size()];
```

Comment:
i'm suprised this code doesn't exist else where for converted meta between string,string map

Reply:
Actually similar logic exist in Schema for writing metadata, I extracted a static write method in FBSerializables for reuse purpose.